plugins/single_plugins/grid.cpp

```diff
@@ -265,24 +288,29 @@ class wayfire_grid : public wf::plugin_interface_t
     void handle_slot(wayfire_view view, const wf_geometry workarea, int slot, bool comes_from_maximize_request)
     {
         wf_geometry target = get_slot_dimensions(slot, workarea);
-        if (view->maximized && view->get_wm_geometry() == target)
+        if (view->tiled_edges && view->get_wm_geometry() == target)
         {
-            return;
+            if (restore_opt_str == restore_opt_default)
```

Comment:
Add braces here. Preferably on the same line as the if/else since both statements are short enough.